### PR TITLE
websocket: make `esp_websocket_client_send_with_opcode` a public API (IDFGH-6937)

### PR DESCRIPTION
**Pull request by gabsuren**

### CODE REVIEW

File: components/esp_websocket_client/include/esp_websocket_client.h [hunk 1]
```diff
@@ -193,6 +194,19 @@ int esp_websocket_client_send_bin(esp_websocket_client_handle_t client, const ch
  */
 int esp_websocket_client_send_text(esp_websocket_client_handle_t client, const char *data, int len, TickType_t timeout);
 
+/**
+ * @brief      Write opcode data to the WebSocket connection
+ *
+ * @param[in]  client  The client
+ * @param[in]  opcode  The opcode
+ * @param[in]  data    The data
+ *
```
Comment (david-cermak) [inline components/esp_websocket_client/include/esp_websocket_client.h:203]: Please complete the docs:
```suggestion
 * @param[in]  len  ....
 * @param[in]  timeout ...
 *
```

File: components/esp_websocket_client/include/esp_websocket_client.h [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #include "esp_err.h"
 #include "esp_event.h"
 #include <sys/socket.h>
+#include "esp_transport_ws.h"
```
Comment (david-cermak) [inline components/esp_websocket_client/include/esp_websocket_client.h:18]: This brings a public dependency to ws-transport.

**Review by david-cermak — CHANGES_REQUESTED** [review]
LGTM, just some nitpicks. As for the dependency, I think it's an acceptable change.

File: components/esp_websocket_client/include/esp_websocket_client.h (hunk 2 above)
Comment (gabsuren) [reply components/esp_websocket_client/include/esp_websocket_client.h:18]: Sorry, does it need any changes or can the conversation be marked as resolved?

**Review by david-cermak — APPROVED** [review]

**Pull request merged**